http/sys_metrics.go

```diff
@@ -47,3 +50,22 @@ func handleMetricsUnauthenticated(core *vault.Core) http.Handler {
 		}
 	})
 }
+
+func handleUnauthenticated(core *vault.Core) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		origBody := new(bytes.Buffer)
+		reader := ioutil.NopCloser(io.TeeReader(r.Body, origBody))
+		r.Body = reader
+		req, _, status, err := buildLogicalRequestNoAuth(core.PerfStandby(), w, r)
+		if err != nil || status != 0 {
+			respondError(w, status, err)
+			return
+		}
+		if origBody != nil {
+			r.Body = ioutil.NopCloser(origBody)
+		}
+
+		resp, err := core.HandleRequest(r.Context(), req)
```

Comment:
On standby nodes this call should fail here: https://github.com/hashicorp/vault/blob/3d0bc374335dfbc6a83d9f4d3fbb468d5957b74f/vault/request_handling.go#L402-L404

The system backend isn't mounted on standby nodes, so this change may be a bit harder than this. You may need to move these handlers into helper functions that are either called directly here or in the system backend

Reply:
Nice catch, thanks!  I had tested at the CLI, but inadequately (I hadn't inspected the resulting debug packages.)

I'm on the fence as to whether my new approach is correct.  Instead of moving the existing handlers as you proposed, I'm using the pprof handlers directly.  Since our sys handlers are very thin wrappers around those that don't currently add anything to what they do, this is technically correct in that the user sees the same behaviour with and without the unauth toggle.  On the other hand, if in future we make enhancements to the sys wrappers, they will diverge.  What do you think?